Add the cloud provider script for the vcenter simulator container

jctanner opened the pull request:
##### SUMMARY
Cloud provider script for the vcsim container

##### ISSUE TYPE

 - Feature Pull Request


##### COMPONENT NAME
vcenter.py

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
devel
```

jctanner committed:
Add the cloud provider script for the vcenter simulator container

jctanner committed:
Add example test for vmware_guest_find

jctanner commented:
bot_status

jctanner committed:
Fix the alias

jctanner committed:
re-add alias

jctanner committed:
Use the new image name

jctanner committed:
Fix image name and path again

jctanner committed:
Use pyvmomi from pip install of distro packages

jctanner commented:
Need to skip py3 tests for now due to https://github.com/ansible/ansible/issues/25984

jctanner committed:
Disable python3 testing for now

jctanner committed:
pip should already exist

jctanner committed:
Add to group1

mattclay commented on test/integration/targets/vmware_guest_find/aliases:
```diff
@@ -0,0 +1,4 @@
+posix/ci/cloud/vcenter
+cloud/vcenter
+skip/python3
+posix/ci/group1
```
This test shouldn't have `posix/ci/group1` since it has `posix/ci/cloud/vcenter`.

jctanner replied:
I only added it as a test to see if shippable would execute the test.

mattclay commented on test/integration/targets/vmware_guest_find/meta/main.yml:
```diff
@@ -0,0 +1,2 @@
+dependencies:
+    - prepare_tests
```
I'd leave this out since `prepare_tests` doesn't do anything. It's a carry-over from the old test infrastructure.

jctanner committed:
Remove dependencies

jctanner committed:
Remove group1 from aliases

mattclay committed:
Clean up vCenter plugin:

- Remove unused imports.
- Remove unused variables.
- Always remove simulator container on exit.
- Change "Vcenter" messages to "vCenter".
- Enable pulling of the simulator image.
- Remove unused methods.
- Add NotImplementedError for unimplemented method.

mattclay committed:
Remove empty meta/main.yml file.

mattclay committed:
Remove commented out pip install.

mattclay committed:
Use vcsim hostname to connect to simulator.

This avoids issues when the simulator reports a different IP
than the one needed by the tests to connect to the simulator.

mattclay committed:
Support local testing w/o --docker on macOS.
